Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -269,7 +270,10 @@ PhysicalParticleContainer::AddGaussianBeam(Real x_m, Real y_m, Real z_m,
             Real y = 0.;
             Real z = distz(mt);
 #endif
-            if (plasma_injector->insideBounds(x, y, z)) {
+            if (plasma_injector->insideBounds(x, y, z)  &&
+                std::abs( x - x_m ) < x_cut * x_rms     &&
```

Comment:
Thanks for this cool PR @MaxThevenet !
It is very nice to read.
But I am still a bit ignorant so I just don't understand very well the default case, i.e. when the input file doesn't define the `x_cut` parameter.
```cpp
std::numeric_limits<amrex::Real>::max();
```
Does this contain the information on `x_rms`?

This is just a question for curiosity, but would it be also useful to define a cut not via a fraction of beam length, but at a fixed position?

Reply:
I think with absolute or relative to RMS is fine, as long as it is explicitly mentioned in the doc. I find relative more practical, so I implemented this one.
By default, the beam is cut for something like `std::abs( x - x_m ) < 1.e308 * x_rms `, i.e., it is not cut at all, so if the user doesn't specify `x_cut` the beam isn't cut.

Reply:
Relative is practical yes.
Oh I see. I was thinking of cutting in regards to the box size, but that would me much more confusing.
The way you did it makes a lot of sense now. Thank you for explaining @MaxThevenet !

Reply:
oh, now I see why you asked that `Inf` question earlier today :)

Reply:
above: `Real weight = q_tot/npart/charge` is better written: `Real const weight = q_tot/(npart*charge);` (and similarly a few lines lower)

all the temporary `x`, `y`, `z` vars can also be `const`-ified.